### ISSUE TITLE
On trade deals, AI values luxuries erratically

### ISSUE THREAD

**Issue by synchronos-t** [LoneGazebo/Community-Patch-DLL#4672]
_1. Mod version (i.e Date - 4/23):_
v. 88 (May 20)

_2. Mod list (if using Vox Populi only, leave blank):_
Only Community Patch of the Vox Populi.

_3. Error description:_ 
The AI players value luxuries very erratically, effectively preventing luxury trade, until at some point they start offering huge amounts of gold for them, giving me an easy win.

I played on King as Rome (standard size & time). I tried to exchange my surplus truffles for any other luxury, but the AIs didn't budge. They insisted on getting my only supply of pearls, which I of course didn't give, as it wouldn't benefit me in any way. No amount of money (GPT) would change their mind. At late game, I even offered up to at least 143 GPT, to no avail.

Later in the game, when the AIs started getting hefty gold surpluses, I then could sell them my truffles for huge GPT offers. They came with an initial offer of giving 2 iron, 2 horse (etc.) and open borders for my truffles, but I could add lots of GPT on the table. The savegame I attached has Siam coming with an offer on the next turn, and I could get 67 GPT (+ 2 iron, 2 horse, open borders) from them. Even later in the game, I could get at least up to 90 GPT. Effectively, they would give all their gold surplus (minus some 2-10 GPT) for my one luxury. This of course made the things insanely easy for me, as I could just buy all the new buildings to my capital. My four truffle surplus resources gave me ~100-300 GPT, as long as I had peace with 2-3 players, so I didn't have to bother with trade routes or anything.

---------------------------
Supporting information:
1. A savegame. Move units, press next turn and wait for Siam to come with an offer. Take as much gold as you can. This is the first time in this game when I got this, but it might have happened earlier; I just maybe didn't try it before this point.
[Augustus Caesar_0216 AD-1530.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/2331799/Augustus.Caesar_0216.AD-1530.zip)

2. A screenshot of the deal on the table:
![20180828175208_1](https://user-images.githubusercontent.com/9976993/44783116-10c94180-ab92-11e8-9f4d-df8054254167.jpg)

3. Later in the game, they turn down my offer of 143 GPT for one meager luxury resource, while I'm on very good terms:
![20180828185149_1](https://user-images.githubusercontent.com/9976993/44783143-28082f00-ab92-11e8-8064-870ba92d52f3.jpg)

**Comment by ilteroi** [comment]
i'll make the assigned value scale with the square root of net gold per turn instead of linearly

**Comment by ricanuck** [comment]
Exactly. I reported this on the forums but was disregarded, and I wasn't even sure if it was a bug. Now I see someone else having the same issue. I stopped playing CP only because of this, as it made the game extremely easy.

@ilteroi : just my impression, but I am not sure the solution is as easy. I don't know the code, but smell something going on related to the trade scores (which in CP only games are disabled)...

**Comment by LoneGazebo** [comment]
@ricanuck "I reported this on the forums but was disregarded, and I wasn't even sure if it was a bug."

If you don't report it here, don't expect it to be picked up for analysis.

**Comment by ricanuck** [comment]
@LoneGazebo : funny, because it was you who told me that "one quirk a bug does not make"... :rofl:

**Comment by LoneGazebo** [comment]
Right, but that doesn't meant you shouldn't report it. Reports can be collected, and then we can see if it is just a quirk or not.

**Comment by ridjack** [comment]
@LoneGazebo At what point do we report things that seem strange but we aren't confident about? If I reported every 'quirk' I'm aware of in VP, you'd have pages and pages of reports from me alone, and I'm confident a large percentage of it would be meaningless.

I'm not asking without purpose, either; I've had a laundry list of trade deal shenanigans that cause me some amount of 'wtf?;' my wife has railed in my general direction on multiple occasions about trade deals that just seem utterly nonsensical. I've brought them up but nobody ever seems to see it as a problem, so I've just assumed there's something I'm not understanding and left it at that.

**Comment by LoneGazebo** [comment]
@ridjack discretion is always nice, but if you repeatedly see something that you feel doesn't look quite right, it can't hurt to report. I'm earnest when I state that I can't patrol the forums looking for reports. I'll occasionally stumble on them but really I need them here. I often find solutions when I cross-reference things.

**Comment by ricanuck** [comment]
I’m with ridjack on this, it’s very hard to find the balance between “reporting everything that seems wrong” and “not being annoying with everything that seems wrong”... so I tend to ask on the forums when in doubt, hoping that someone would recognize my “claim”, and if so, then I know it’s not me, and report it. The last thing I want is to fill your git with stuff so that it becomes impossible to manage.

**Comment by ricanuck** [comment]
see? always this reaction man... that also plays a part in not being willing to report something...

**Comment by LoneGazebo** [comment]
>see? always this reaction man... that also plays a part in not being willing to report something...

Not sure what you mean. Two+ people reported it, ilteroi took a look, and there's a potential fix inbound. You want a greeting card?

**Comment by ricanuck** [comment]
yes please, with a cheque.

**Comment by ryanmusante** [comment]
@ricanuck Dude, you should be sending Gazebo a cheque. He does this in his free time as a labor of love. There is no payment from Firaxis. Be appreciative and relax.

**Comment by ridjack** [comment]
@LoneGazebo I'm so sorry. I just wanted to ask a simple clarifying question, I didn't want it to become drama. :|

**Comment by ricanuck** [comment]
“Dude”, you should learn to recognize humor even if it’s not to your taste, and as for appreciating the labor of modders... duh!! I’m one of them.

**Comment by ryanmusante** [comment]
@ricanuck I realize that now, my bad.

**Comment by ricanuck** [comment]
no problem... the downs of the Internet (not being able to see the other person’s face and body language).
